lib/transform.js

```diff
@@ -24,64 +25,41 @@ function addAnchor(mode, header) {
   return header;
 }
 
+function isString(y) {
+  return typeof y === 'string';
+}
 
-function getHashedHeaders (lines, maxHeaderLevel) {
-  var inCodeBlock = false
-    , lineno = 0;
 
-  // Turn all headers into '## xxx' even if they were '## xxx ##'
-  function normalize(header) {
-    return header.replace(/[ #]+$/, '');
+function getMarkdownHeaders (lines, maxHeaderLevel) {
+  function extractText (header) {
+    return header.children
+      .filter(function (x) {
+        return x.type !== 'Image';
+      })
+      .map(function (x) {
+        if (x.type === 'Link') {
+          return extractText(x);
+        }
```

Comment:
Oh hmm, I was thinking none of those should be kept. I guess I'll defer to @thlorenz then.



> Should we handle the case there is html in a header?

I don't think that many people have run into this issue. Mostly people have additional markdown within their headers. We can potentially address this if at some point we decide to solidify our story around "parsing" HTML in the first place.